frontend/dashboard/components/MakeCopyModal/MakeCopyModal.tsx

```diff
@@ -1,126 +1,99 @@
-import type { MouseEvent, ChangeEvent } from 'react';
-import React, { useState } from 'react';
-import { StudioSpinner } from '@studio/components';
-import { AltinnPopoverSimple } from 'app-shared/components/molecules/AltinnPopoverSimple';
-import type { PopoverOrigin } from '@mui/material';
-import { APP_DEVELOPMENT_BASENAME } from 'app-shared/constants';
-import { validateRepoName } from '../../utils/repoUtils';
-import { Textfield } from '@digdir/design-system-react';
+import React, { forwardRef, useState } from 'react';
+import { StudioModal } from '@studio/components';
+import { Heading } from '@digdir/design-system-react';
 import classes from './MakeCopyModal.module.css';
-import { SimpleContainer } from 'app-shared/primitives';
 import { useTranslation } from 'react-i18next';
 import { useCopyAppMutation } from 'dashboard/hooks/mutations/useCopyAppMutation';
 import type { AxiosError } from 'axios';
 import { ServerCodes } from 'app-shared/enums/ServerCodes';
+import { useUserQuery } from 'app-shared/hooks/queries';
+import { useOrganizationsQuery } from '../../hooks/queries';
+import { NewApplicationForm } from '../NewApplicationForm';
+import { type NewAppForm } from '../../types/NewAppForm';
+import { PackagesRouter } from 'app-shared/navigation/PackagesRouter';
 
-export interface IMakeCopyModalProps {
-  anchorEl: HTMLElement;
-  handleClose: (event?: MouseEvent<HTMLElement>) => void;
+export type MakeCopyModalProps = {
+  open: boolean;
+  onClose: () => void;
   serviceFullName: string;
-}
-
-const transformAnchorOrigin: PopoverOrigin = {
-  vertical: 'center',
-  horizontal: 'center',
 };
 
-export const MakeCopyModal = ({ anchorEl, handleClose, serviceFullName }: IMakeCopyModalProps) => {
-  const {
-    mutate: copyAppMutate,
-    isPending: isCopyAppPending,
-    isError: hasCopyAppError,
-  } = useCopyAppMutation({
-    hideDefaultError: (error: AxiosError) => error?.response?.status === ServerCodes.Conflict,
-  });
-  const [repoName, setRepoName] = useState<string>('');
-  const [errorMessage, setErrorMessage] = useState<string>(null);
+export const MakeCopyModal = forwardRef<HTMLDialogElement, MakeCopyModalProps>(
+  ({ open, onClose, serviceFullName }, ref) => {
+    const { data: user } = useUserQuery();
+
+    const { data: organizations } = useOrganizationsQuery();
+    const {
+      mutate: copyAppMutate,
+      isPending: isCopyAppPending,
+      isSuccess: isCopyAppSuccess,
+    } = useCopyAppMutation({
+      hideDefaultError: (error: AxiosError) => error?.response?.status === ServerCodes.Conflict,
+    });
+
+    const [formError, setFormError] = useState<NewAppForm>({ org: '', repoName: '' });
 
-  const { t } = useTranslation();
+    const { t } = useTranslation();
 
-  const handleClone = async () => {
-    if (validAppName()) {
+    const createClonedRepo = async (newAppForm: NewAppForm) => {
```

Comment:
We can remove the async keyword from here. 😊 
```suggestion
    const createClonedRepo = (newAppForm: NewAppForm) => {
```

Reply:
I'm not sure if we can actually 😄 The function is used in the `<NewApplicationForm />` component, and is being called using `async-await`: 
![image](https://github.com/Altinn/altinn-studio/assets/133344438/c18418ad-5427-4f44-a7f8-712bb4483f31)

Reply:
snap, I missed that one! 🙈